### PR TITLE
[CONSUL-606] Update container_init.go

### PR DESCRIPTION
**Pull request by joselo85**
# Description:
Updated container_init.go to use pod OS detection.

### CODE REVIEW

File: control-plane/connect-inject/webhook/container_init.go [hunk 1]
```diff
@@ -38,6 +38,24 @@ type initContainerCommandData struct {
 // containerInit returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
 // so that it can save the proxy service id to the shared volume and boostrap Envoy with the proxy-id.
 func (w *MeshWebhook) containerInit(namespace corev1.Namespace, pod corev1.Pod, mpi multiPortInfo) (corev1.Container, error) {
+	var volumeMountPath, consulAddress, imageConsulK8s, initContainerCommandInterpreter, initContainerCmd string
+
+	if isWindows(pod) {
+		volumeMountPath = "C:\\consul\\connect-inject"
+		// Windows resolves DNS addresses differently. Read more: https://github.com/hashicorp-education/learn-consul-k8s-windows/blob/main/WindowsTroubleshooting.md#encountered-issues
+		consulAddress, _, _ = strings.Cut(w.ConsulAddress, ".")
+		// TODO
```
Comment (ezfepo) [inline control-plane/connect-inject/webhook/container_init.go:47]:
```suggestion
		// TODO -> Update webhook struct CONSUL-599
```

File: control-plane/connect-inject/webhook/container_init.go [hunk 2]
```diff
@@ -38,6 +38,24 @@ type initContainerCommandData struct {
...
+		imageConsulK8s = "windows consul k8s control plane image"
+		initContainerCommandInterpreter = "sh"
+		initContainerCmd = initContainerCommandTplWindows
+	} else {
+		volumeMountPath = "/consul/connect-inject"
```
Comment (ezfepo) [inline control-plane/connect-inject/webhook/container_init.go:52]: rename variable to 
_connectInjectDir_

File: control-plane/connect-inject/webhook/consul_dataplane_sidecar.go [hunk 3]
```diff
@@ -61,6 +61,7 @@ func (w *MeshWebhook) consulDataplaneSidecar(namespace corev1.Namespace, pod cor
 
 	// Assign values to the variables depending on the OS
 	if isWindows(pod) {
+		// TODO -> Update webhook struct CONSUL-599
 		dataplaneImage = "windows consul dataplane image"
```
Comment (ezfepo) [inline control-plane/connect-inject/webhook/consul_dataplane_sidecar.go:65]: rename variable to
ImageConsulDataplane

File: control-plane/connect-inject/webhook/container_init.go [hunk 4]
```diff
@@ -38,6 +38,24 @@ type initContainerCommandData struct {
...
+	var volumeMountPath, consulAddress, imageConsulK8s, initContainerCommandInterpreter, initContainerCmd string
+
+	if isWindows(pod) {
+		volumeMountPath = "C:\\consul\\connect-inject"
+		// Windows resolves DNS addresses differently. Read more: https://github.com/hashicorp-education/learn-consul-k8s-windows/blob/main/WindowsTroubleshooting.md#encountered-issues
+		consulAddress, _, _ = strings.Cut(w.ConsulAddress, ".")
+		// TODO
+		imageConsulK8s = "windows consul k8s control plane image"
+		initContainerCommandInterpreter = "sh"
+		initContainerCmd = initContainerCommandTplWindows
```
Comment (ezfepo) [inline control-plane/connect-inject/webhook/container_init.go]: rename variable to 
_initContainerCommandTpl_

File: control-plane/connect-inject/webhook/container_init.go [hunk 5]
```diff
@@ -38,6 +38,24 @@ type initContainerCommandData struct {
...
+	} else {
+		volumeMountPath = "/consul/connect-inject"
+		consulAddress = w.ConsulAddress
+		imageConsulK8s = w.ImageConsulK8S
+		initContainerCommandInterpreter = "/bin/sh"
+		initContainerCmd = initContainerCommandTpl
```
Comment (ezfepo) [inline control-plane/connect-inject/webhook/container_init.go]: rename variable to 
_initContainerCommandTplLinux_

File: control-plane/connect-inject/webhook/consul_dataplane_sidecar.go (hunk 3 above)
Comment (ezfepo) [inline control-plane/connect-inject/webhook/consul_dataplane_sidecar.go:65]: rename variable to
_ImageConsulDataplane_

File: control-plane/connect-inject/webhook/container_init.go (hunk 1 above)
Comment (ezfepo) [inline control-plane/connect-inject/webhook/container_init.go]:
```suggestion
		// TODO -> Update webhook struct CONSUL-599
```

File: control-plane/connect-inject/webhook/container_init.go (hunk 2 above)
Comment (ezfepo) [inline control-plane/connect-inject/webhook/container_init.go]: rename variable to
_connectInjectDir_

File: control-plane/connect-inject/webhook/container_init.go (hunk 4 above)
Comment (joselo85) [reply control-plane/connect-inject/webhook/container_init.go:50]: That name was already taken for the original constant in line 319. That's why I went with this var name.

**Review by ezfepo — APPROVED** [review]

**Pull request merged**